### PR TITLE
feat(form): input password with toggle

### PR DESCRIPTION
**Pull request by smouillour**
**What is the problem this PR is trying to solve?**
Having the ability to switch a password field to field text

**What is the chosen solution to this problem?**
Create a new widget for a password field with a toggle

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR
Frontify : https://company-57688.frontify.com/document/92132#/ui-controls/input-field/password-field

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by smouillour** [comment]
the exemple is in Form/Simple

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { renderActions } from '../../Form';
+import theme from './TogglePasswordWidget.scss';
+
+export default class TogglePasswordWidget extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onToggle = this.onToggle.bind(this);
+		this.state = {
+			type: props.type?props.type:'password',
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: state should not comes from props.
I would prefer this.state = { type: 'password' }
by default this seems far better

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	onToggle() {
+		this.setState({ type: this.state.type === 'text' ? 'password' : 'text' });
+	}
+
+	handleActionClick(onClick) { // eslint-disable-line class-methods-use-this
+		if (onClick) {
+			return () => onClick();
+		}
+		return () => {
+		};
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: this looks weird.
handle doesn't not handle the click it returns a function.
getOnClick ?

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	render() {
+		const propsWithToggleType = {
+			...this.props,
+			type: this.state.type,
+		};
+		return (
+			<div className={classNames(theme['btn-group'], 'btn-group')}>
+				<Password {...propsWithToggleType} />
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: I would prefere only one spread.
`...this.props type={this.state.type}`

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
+.btn-group {
+    width: 100% !important;
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: really needed ?

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    &>button{
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: lint : add space here

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.test.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+		const wrapper = shallow(
+			<TogglePasswordWidget
+				value={value}
+				type={type}
+			/>
+		);
+
+		// then
+		expect(toJson(wrapper)).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.test.js]: wrapper.getNode() is working well also

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+				{renderActions([{
```
Comment (acateland) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: please extract first parameter construction from jsx declaration

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.test.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import TogglePasswordWidget from './TogglePasswordWidget';
+
+describe('TogglePasswordWidget', () => {
```
Comment (acateland) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.test.js]: please add the test checking if interaction work correctly, check that a t first is renderered a password field, simulate click and check that it is now renderered as an input

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss (hunk 4 above)
Comment (smouillour) [reply packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: if i don't force it the field is reduce at 200px and i don't find why.

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss [hunk 9]
```diff
@@ -0,0 +1,23 @@
+.btn-group {
```
Comment (frassinier) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: maybe you can use specific class for these overrides

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss (hunk 5 above)
Comment (frassinier) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: This `&` is useless

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js (hunk 2 above)
Comment (smouillour) [reply packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: same code as Form.js (handleActionClick(onClick))

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+	onToggle() {
+		this.setState({ type: this.state.type === 'text' ? 'password' : 'text' });
+	}
+
+	handleActionClick(onClick) { // eslint-disable-line class-methods-use-this
+		if (onClick) {
+			return () => onClick();
+		}
+		return () => {};
+	}
```
Comment (jmfrancois) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js]: ok so this the case on every other widgets

File: packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss [hunk 11]
```diff
@@ -0,0 +1,23 @@
+.password-group {
+    // sass-lint:disable no-important
+    width: 100% !important;
```
Comment (jare-talend) [inline packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.scss]: Could you remove this important ?

**Pull request closed without merging**